File: flightbooking/springapp/src/main/java/com/examly/springapp/service/BookingService.java
package com.examly.springapp.service;

import com.examly.springapp.model.Booking;

import java.util.List;

public interface BookingService {
    Booking createBooking(Booking booking);
    Booking getBookingById(Long id);
    List<Booking> getAllBookings();
    Booking updateBooking(Long id, Booking booking); // New method for updating bookings
    List<Booking> getBookingsByUserId(int userId);
    
}

File: flightbooking/angularapp/src/app/service/booking.service.ts
// booking.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Booking } from '../models/booking.model'; 
import { apiUrl } from '../apiconfig'; 

@Injectable({
  providedIn: 'root'
})
export class BookingService {
  public baseUrl = apiUrl;
 // Replace with your backend URL

 public apiUrl = `${this.baseUrl}/api/bookings`;

  constructor(private http: HttpClient) { }

  createBooking(booking: Booking): Observable<Booking> {
    return this.http.post<Booking>(this.apiUrl, booking);
  }

  getBookingById(id: number): Observable<Booking> {
    return this.http.get<Booking>(`${this.apiUrl}/${id}`);
  }

  getAllBookings(): Observable<Booking[]> {
    return this.http.get<Booking[]>(this.apiUrl);
  }

  updateBooking(id: number, booking: Booking): Observable<Booking> {
    return this.http.put<Booking>(`${this.apiUrl}/${id}`, booking);
  }

  deleteBooking(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
